suite-native/device/src/middlewares/deviceMiddleware.ts

```diff
@@ -78,11 +84,32 @@ export const prepareDeviceMiddleware = createMiddlewareWithExtraDeps(
 
         switch (action.type) {
             case DEVICE.CONNECT:
-            case DEVICE.CONNECT_UNACQUIRED:
+            case DEVICE.CONNECT_UNACQUIRED: {
                 if (isUsbDeviceConnectFeatureEnabled) {
                     dispatch(selectDeviceThunk(action.payload));
                 }
+
+                const {
+                    device: { features, mode },
+                } = action.payload;
```

Comment:
What about this?
```suggestion
                const { device } = action.payload;
                const { features, mode } = device;
```

That way you could replace three usages of `action.payload.device`.

Reply:
fixed and squashed